### PR TITLE
[Static references] LPS-182376 Remove static references in modules site-navigation, social and wiki

### PR DESCRIPTION
**Pull request by RenanUmV**
[LPS-182376](https://issues.liferay.com/browse/LPS-182376)

### CODE REVIEW

**Comment by RenanUmV** [comment]
ci:test:wiki

**Comment by Kvale1733** [comment]
@RenanUmV Regression found. Seems the portal is completely broken with these changes. There are a lot of startup errors.
[Log](https://gist.github.com/Kvale1733/0cbea95bde0c4e82d8d3cc2e9db531d1)

Most of the UI is broken and there is no response when clicking any  of it. The welcome page also shows "An unexpected system error occurred".

File: modules/apps/wiki/wiki-web/src/main/java/com/liferay/wiki/web/internal/display/context/DLMimeTypeDisplayContextUtil.java [hunk 1]
```diff
@@ -15,43 +15,28 @@
...
-	protected void deactivate() {
-		_dlMimeTypeDisplayContextUtil = null;
-	}
-
-	private static DLMimeTypeDisplayContextUtil _dlMimeTypeDisplayContextUtil;
-
-	@Reference
-	private DLMimeTypeDisplayContext _dlMimeTypeDisplayContext;
+	private static final DLMimeTypeDisplayContext _dlMimeTypeDisplayContext =
+		DLMimeTypeDisplayContextUtil._dlMimeTypeDisplayContextSnapshot.get();
```
Comment (rafaprax) [inline modules/apps/wiki/wiki-web/src/main/java/com/liferay/wiki/web/internal/display/context/DLMimeTypeDisplayContextUtil.java:37]: Hey @RenanUmV, you do not need this field `_dlMimeTypeDisplayContext`. You just need to use the snapshot in the methods implementations.

Thx

File: modules/apps/wiki/wiki-engine-input-editor-common/src/main/java/com/liferay/wiki/engine/input/editor/common/internal/util/WikiEngineInputEditorCommonComponentProvider.java [hunk 2]
```diff
@@ -34,25 +30,14 @@ public class WikiEngineInputEditorCommonComponentProvider {
...
-		_wikiEngineInputEditorCommonComponentProvider = null;
+		return _servletContextSnapshot.get();
 	}
 
+	private static final Snapshot<ServletContext> _servletContextSnapshot =
+		new Snapshot<>(
+			WikiEngineInputEditorCommonComponentProvider.class,
+			ServletContext.class);
 	private static WikiEngineInputEditorCommonComponentProvider
 		_wikiEngineInputEditorCommonComponentProvider;
```
Comment (rafaprax) [inline modules/apps/wiki/wiki-engine-input-editor-common/src/main/java/com/liferay/wiki/engine/input/editor/common/internal/util/WikiEngineInputEditorCommonComponentProvider.java:42]: Hey @RenanUmV, now the field `_wikiEngineInputEditorCommonComponentProvider` is never initialized. By the way, you do not need this more. 

Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by rafaprax** [comment]
Hey @RenanUmV, see my comments and resend the pr. Thx

**Pull request closed without merging**